orchagent/bfdorch.cpp

```diff
@@ -0,0 +1,412 @@
+    attr.value.s32 = bfd_session_type;
+    attrs.emplace_back(attr);
+    fvVector.emplace_back("type", session_type_loopup.at(bfd_session_type));
+
+    attr.id = SAI_BFD_SESSION_ATTR_LOCAL_DISCRIMINATOR;
+    attr.value.u32 = bfd_gen_id();
+    attrs.emplace_back(attr);
+
+    attr.id = SAI_BFD_SESSION_ATTR_UDP_SRC_PORT;
+    attr.value.u32 = bfd_src_port();
+    attrs.emplace_back(attr);
+
+    attr.id = SAI_BFD_SESSION_ATTR_REMOTE_DISCRIMINATOR;
+    attr.value.u32 = 0;
+    attrs.emplace_back(attr);
+
+    attr.id = SAI_BFD_SESSION_ATTR_BFD_ENCAPSULATION_TYPE;
+    attr.value.s32 = encapsulation_type;
+    attrs.emplace_back(attr);
+
+    attr.id = SAI_BFD_SESSION_ATTR_IPHDR_VERSION;
+    attr.value.u8 = src_ip.isV4() ? 4 : 6;
+    attrs.emplace_back(attr);
+
+    if (!src_ip_provided)
+    {
+        SWSS_LOG_ERROR("Failed to create BFD session %s because source IP is not provided", key.c_str());
+        return false;
+    }
+    attr.id = SAI_BFD_SESSION_ATTR_SRC_IP_ADDRESS;
+    copy(attr.value.ipaddr, src_ip);
+    attrs.emplace_back(attr);
+    fvVector.emplace_back("local_addr", src_ip.to_string());
+
+    attr.id = SAI_BFD_SESSION_ATTR_DST_IP_ADDRESS;
+    copy(attr.value.ipaddr, peer_address);
+    attrs.emplace_back(attr);
+
+    attr.id = SAI_BFD_SESSION_ATTR_MIN_TX;
+    attr.value.u32 = tx_interval * 1000;
+    attrs.emplace_back(attr);
+    fvVector.emplace_back("tx_interval", to_string(tx_interval));
+
+    attr.id = SAI_BFD_SESSION_ATTR_MIN_RX;
+    attr.value.u32 = rx_interval * 1000;
+    attrs.emplace_back(attr);
+    fvVector.emplace_back("rx_interval", to_string(rx_interval));
+
+    attr.id = SAI_BFD_SESSION_ATTR_MULTIPLIER;
+    attr.value.u8 = multiplier;
+    attrs.emplace_back(attr);
+    fvVector.emplace_back("multiplier", to_string(multiplier));
+
+    if (multihop)
+    {
+        attr.id = SAI_BFD_SESSION_ATTR_MULTIHOP;
+        attr.value.booldata = true;
+        attrs.emplace_back(attr);
+        fvVector.emplace_back("multihop", "true");
+    }
+    else
+    {
+        fvVector.emplace_back("multihop", "false");
+    }
+
+    if (alias != "default")
+    {
+        attr.id = SAI_BFD_SESSION_ATTR_HW_LOOKUP_VALID;
+        attr.value.booldata = false;
+        attrs.emplace_back(attr);
+
+        Port port;
+        if (!gPortsOrch->getPort(alias, port))
+        {
+            SWSS_LOG_ERROR("Failed to locate port %s", alias.c_str());
+            return false;
+        }
+
+        attr.id = SAI_BFD_SESSION_ATTR_PORT;
+        attr.value.oid = port.m_port_id;
+        attrs.emplace_back(attr);
+
+        attr.id = SAI_BFD_SESSION_ATTR_SRC_MAC_ADDRESS;
+        memcpy(attr.value.mac, port.m_mac.getMac(), sizeof(sai_mac_t));
+        attrs.emplace_back(attr);
+
+        if (!dst_mac_provided)
+        {
+            SWSS_LOG_ERROR("Failed to create BFD session %s: destination MAC address required when hardware lookup not valid",
+                            key.c_str());
+            return false;
```

Comment:
return `true` as no need to retry. Suggest check all return codes in this function

Reply:
Updated return value to `true` and checked other return codes